src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -107,27 +110,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     // creates arraylist and starts query
-    ArrayList<String> groceryList = new ArrayList<String>();
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
     Query groceryQuery = new Query(GROCERY)
       .setFilter(new Query.FilterPredicate(FAMILY_ID, Query.FilterOperator.EQUAL, familyID));   
     PreparedQuery familyGrocery= datastore.prepare(groceryQuery);
-    groceryList = checkGroceries(familyGrocery);   
+    groceryList = checkGroceries(familyGrocery, userEmail);   
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(groceryList));
     }
 
-  // returns items from the query that match the users familyID
-  private ArrayList<String> checkGroceries(PreparedQuery familyGrocery) {
-    ArrayList<String> groceryList = new ArrayList<String>();
+  // returns items from the query that match the users familyID in the Grocery object    
+  private ArrayList<Grocery> checkGroceries(PreparedQuery familyGrocery, String userEmail) {
+    boolean match;
+    long timestamp = System.currentTimeMillis();
+    System.out.println(timestamp);
+    ArrayList<Grocery> groceryList = new ArrayList<>();
     for (Entity entity : familyGrocery.asIterable()) {
-        String groceryOutput;
-            String groceryItem = (String) entity.getProperty("grocery");
-            String memberEmail = (String) entity.getProperty("assignEmail");
-            groceryOutput = groceryItem + " assigned to: " + memberEmail;
-            groceryList.add(groceryOutput);
+        String groceryItem = (String) entity.getProperty(GROCERY);
+        String assignEmail = (String) entity.getProperty("assignEmail");
+        long id = entity.getKey().getId();     
+        if(userEmail.equals(assignEmail)) {
+            match = true;
```

Comment:
Hmm, rather than match, maybe change this to something like `isAssignedToCurrentUser` or `belongsToCurrentUser`?

match is an ambiguous word (we don't know which thing it's matching to).

There are two cases you are handling right now:
1. is assigned to no one
2. is assigned to the current user (which is hard to get, because you don't have the user's email on the client side)

(1) is something you can figure out on the client side based on the assignEmail that is returned with the grocery. So the only additional piece of information you need to represent is (2)/